docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -2501,3 +2501,76 @@ declare module "@material-ui/private-theming" {
   interface DefaultTheme extends Theme {}
 }
 ```
+
+### makeStyles - TypeError: Cannot read property 'drawer' of undefined
+
+This error occurs when calling `useStyles` (result of `makeStyles`) or `withStyles` outside of `<ThemeProvider>` scope like this:
+
+```js
+import * as React from 'react';
+import { ThemeProvider, createTheme } from '@material-ui/core/styles';
+import makeStyles from '@material-ui/styles/makeStyles';
+import Card from '@material-ui/core/Card';
+import CssBaseline from '@material-ui/core/CssBaseline';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    backgroundColor: theme.palette.primary.main,
+    color: theme.palette.common.white,
+  },
+}));
+
+const theme = createTheme();
+
+function App() {
+  const classes = useStyles(); // ❌ called outside of ThemeProvider
+  return (
+    <ThemeProvider theme={theme}>
+      <CssBaseline />
+      <Card className={classes.root}>...</Card>
+    </ThemeProvider>
+  );
+}
+
+export default App;
+```
+
+One way to fix this is to create `withThemeProvider` HOC (higher order component):
```

Comment:
I was expecting more suggestions when I read "One way to fix it..." :) Do we have more suggestions?

I don't think our suggestion should be using a HOC, we can simply say that the root App component should only have the `ThemeProvider` and then the rest of the components inside it's tree can use the `useStyles` hooks. We can reformat the example to look something like:

```jsx
import * as React from 'react';
import { ThemeProvider } from '@material-ui/core/styles';
import makeStyles from '@material-ui/styles/makeStyles';
import Card from '@material-ui/core/Card';
import CssBaseline from '@material-ui/core/CssBaseline';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.common.white,
  },
}));

const theme = createTheme();
function AppContent() {
  const classes = useStyles(); // ✅ This is safe because App is wrapping the AppContent with <ThemeProvider>
  return <Card className={classes.root}>...</Card>;
}

const App = (props) => {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <AppContent {...props} />
    </ThemeProvider>
  );
};

export default App;
```

Or maybe this can be our first suggestion, and the HOC second.

Reply:
fixed.